includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -851,6 +851,13 @@ public function process_refund( $order_id, $amount = null, $reason = '' ) {
 		}
 
 		if ( $reason ) {
+			// Trim the refund reason to a max of 499 characters.
+			if ( strlen( $reason ) >= 500 ) {
+				$reason = substr( $reason, 0, 450 );
+				// Add some explainer text indicating where to find the full refund reason.
```

Comment:
I believe we could use `strlen( $reason ) > 500` instead of `strlen( $reason ) >= 500`, so that we use exactly the maximum limit Stripe allows us. I've tested with a 500 characters long refund reason, and it worked.

Perhaps we could change that first comment as well to make it clear that we're doing this because of Stripe's hard limit on metadata values, so other developers can understand the reason for this condition right away.

From what I've tested, this fix works great, but I found an edge case, which may not be that critical, but I'm used to face this type of problem because I speak Portuguese: `substr` unfortunately counts for bytes instead of characters. If it happens that the refund reason gets cut right on an accented character or any multi-byte character, the character gets corrupted, and for some reason the refund doesn't go through.

You can test it with the following refund reason:

```
Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Loreé dsadsadasdsadsadsadsadsadasdasdasdasdasdasdasdasda
```

The above text is 501 characters long, and it will be trimmed right in the "é" character, which is the 450th character.

As fallback for such case, we could maybe use [mb_substr](https://www.php.net/manual/pt_BR/function.mb-substr.php), which [doesn't come by default](https://www.php.net/manual/en/mbstring.installation.php) with PHP, but I believe most hosts should have it.

It seems like Automattic always [checks if this function exists](https://github.com/Automattic/jetpack-threat-library-cegfault/blob/2742d49cabb98add6b07922786a8e8fe1e01bdbf/false-positives/top-plugins/the-events-calendar/common/src/functions/multibyte.php#L82) first. And you could do the same here:

```php
$reason = function_exists( 'mb_substr' ) ? mb_substr( $reason, 0, 450 ) : substr( $reason, 0, 450 );
```

For `strlen` it's not multi-byte either, but I believe there's no problem in using it, because it will always count for more characters than the actual number of characters if there is any accent, so that's not a problem.

Reply:
> I believe we could use strlen( $reason ) > 500 instead of strlen( $reason ) >= 500, so that we use exactly the maximum limit Stripe allows us. I've tested with a 500 characters long refund reason, and it worked.

Ah, good! I did think of that, but considering that the error messages states that the reason "must be a string under 500 characters", I chose to play it safe and cut it off at 499, since technically speaking a 500 character string is not "a string under 500 characters."

Anyway, yeah, I'll change the value. :) Thanks for testing it!

> Perhaps we could change that first comment as well to make it clear that we're doing this because of Stripe's hard limit on metadata values, so other developers can understand the reason for this condition right away.

Will do!

> As fallback for such case, we could maybe use `mb_substr`, which doesn't come by default with PHP, but I believe most hosts should have it.

I thought of this as well, but then looking at the code for this extension, we don't use `mb_substr` at all, just `substr`, so that's what I went with. ¯\_(ツ)_/¯ Happy to check for it first and fall back to plain old `substr` though.